iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/SubnetworkImpl.java

```diff
@@ -696,20 +741,103 @@ public Network detach() {
      */
     @Override
     public boolean isDetachable() {
-        //TODO subnetworks API
-        throw new UnsupportedOperationException("Not yet implemented");
+        return checkDetachable(getBoundaryElements(), false);
+    }
+
+    private boolean checkDetachable(Set<Identifiable<?>> boundaryElements, boolean throwsException) {
+        if (parent.getVariantManager().getVariantArraySize() != 1) {
+            if (throwsException) {
+                throw new PowsyblException("Detaching from multi-variants network is not supported");
+            }
+            return false;
+        }
+        if (boundaryElements.stream().anyMatch(Predicate.not(SubnetworkImpl::isSplittable))) {
+            if (throwsException) {
+                throw new PowsyblException("Some un-splittable equipments prevent the subnetwork to be detached");
+            }
+            return false;
+        }
+        return true;
+    }
+
+    private static boolean isSplittable(Identifiable<?> identifiable) {
+        return identifiable.getType() == IdentifiableType.TIE_LINE;
     }
 
     @Override
     public Set<Identifiable<?>> getBoundaryElements() {
-        //TODO subnetworks API
-        throw new UnsupportedOperationException("Not yet implemented");
+        return getPotentialBoundaryElements()
+                .filter(this::isBoundaryElement)
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Return all the potential boundary elements: elements defined in the current subnetwork or in the parent network
+     * and which type corresponds to a "linking" element (branch, transformer, ...).
+     *
+     * @return a {@link Stream} of the potential boundary elements
+     */
+    private Stream<Identifiable<?>> getPotentialBoundaryElements() {
+        Stream<Line> lines = parent.getLineStream();
+        Stream<TwoWindingsTransformer> twoWindingsTransformerStream = parent.getTwoWindingsTransformerStream();
```

Comment:
To check: two and three windings transformers cannot be outside of a substation. A substation cannot be shared into two subnetworks.